packages/stack-shared/src/utils/urls.tsx

```diff
@@ -225,6 +226,31 @@ import.meta.vitest?.test("isLocalhost", ({ expect }) => {
   expect(isLocalhost("")).toBe(false);
 });
 
+export function extractBaseDomainFromHost(host: string): string {
+  const hostWithoutPort = host.split(":")[0];
+  if (!hostWithoutPort) {
+    return hostWithoutPort;
+  }
+  if (hostWithoutPort === "localhost") {
+    return hostWithoutPort;
+  }
+  const parsed = parse(hostWithoutPort, { allowPrivateDomains: true });
+  if (parsed.isIp) {
+    return hostWithoutPort;
+  }
+  const domain = getDomain(hostWithoutPort, { allowPrivateDomains: true });
+  return domain ?? hostWithoutPort;
+}
+import.meta.vitest?.test("extractBaseDomainFromHost", ({ expect }) => {
+  expect(extractBaseDomainFromHost("app.example.com")).toBe("example.com");
+  expect(extractBaseDomainFromHost("sub.app.example.com")).toBe("example.com");
+  expect(extractBaseDomainFromHost("example.com")).toBe("example.com");
+  expect(extractBaseDomainFromHost("localhost:3000")).toBe("localhost");
+  expect(extractBaseDomainFromHost("127.0.0.1")).toBe("127.0.0.1");
+  expect(extractBaseDomainFromHost("127.0.0.1:3000")).toBe("127.0.0.1");
+  expect(extractBaseDomainFromHost("app.example.co.uk")).toBe("example.co.uk");
+});
```

Comment:
i think we won't need this function, and given the security issues, I'd remove it

```suggestion
```